lib/game_stats.rb

```diff
@@ -1,4 +1,35 @@
 
 
 module GameStats
+
+    def count_of_games_by_season 
+        games_per_season = Hash.new(0)
+        @games.each do |game|
+            if games_per_season.has_key?(game.season_id)
+                games_per_season[game.season_id] += 1
+            else
+                games_per_season[game.season_id] = 1
+            end
+        end
+        games_per_season
+    end
+
+    def average_goals_per_game
+        average_of_games = []
+        @games.each do |game|
+            average_of_games << game.away_goals.to_f
+            average_of_games << game.home_goals.to_f
+        end
+        average_of_games = average_of_games.sum / average_of_games.size
+        average_of_games.round(2)
+    end
+
+    def average_goals_by_season
+        average_goals_per_season = {}
+        @games.each do |game|
+            if average_goals_per_season.has_key?(game.season_id)
```

Comment:
same thing as Chee commented above, I don't think you need the if statement because if there isn't an existing key, ruby knows to create a key using (game.season_id) the first time it iterates through one of the game objects

Reply:
Oh man! That is really good to know thanks for the feedback.